e2e_tests/e2e.test.ts

```diff
@@ -271,6 +271,16 @@ describe("webdriver", () => {
         }).finally(() => killDriver(driver))
     })
 
+    test("`:tabopen foo/<CR>` opens http://foo.", async () => {
+        const driver = await getDriver()
+        return newTabWithoutChangingOldTabs(driver, async () => {
+            await sendKeys(driver, ":tabopen foo/<CR>")
+        }).then(async ([newtab, _]) => {
+            expect(newtab.active).toEqual(true)
+            await driver.wait(untilTabUrlMatches(driver, newtab.id, "http://foo"), 10000)
+        }).finally(() => killDriver(driver))
+    })
+
```

Comment:
Thanks for this PR! I thought I had added this feature years ago :)


Annoyingly (as you discovered) the end-to-end tests are extremely fragile. At the moment they seem totally broken, but even when they aren't, they often take 10-20 attempts to succeed on CI. It gets worse with every test we add. So please don't add any new E2E tests :)

You should be able to add a unit test by making a file with the extension `.test.ts`, which should automatically swap `browser` out for a fake one via this package https://github.com/clarkbw/jest-webextension-mock - but we don't seem to be using it anywhere else so it's possible that it doesn't work.

From my perspective, I don't really mind if the code doesn't have a test. But I understand I am a bit of a heretic for having such views.

Reply:
I removed the e2e test.

I tried to add some unit tests, but it seems like the test environment isn't mocking enough to allow `webext.ts` to be included (`getContext` is easy enough, but I stopped at `this.addEventListener.bind` because it doesn't seem to be part of the source).  
Maybe moving `queryAndURLwrangler` to a different file would allow it to work.  I could work on that in a follow-up PR, if you'd like.  I have the impression the function could be refactored and unit tests would be a nice safety net for anyone working on that; otherwise I'm normally also a bit of a heretic.

Reply:
I don't think moving it to a different file is worth your time, but if you want to do it, go ahead.

If you're in the mood to fight with tests, I'd much rather know why the e2e tests fail all the time at `:rssexec`. I had a go looking at it in #4990